src/main/java/io/confluent/examples/streams/microservices/OrderDetailsService.java

```diff
@@ -66,23 +67,31 @@ private void startService(String bootstrapServers) {
     try {
       Map<TopicPartition, OffsetAndMetadata> consumedOffsets = new HashMap<>();
       consumer.subscribe(singletonList(Topics.ORDERS.name()));
-      producer.initTransactions();
+      // Commenting out configuration for Exactly Once Semantics (EOS)
+      // because currently EOS is not supported with Confluent Monitoring Interceptors 
+      //producer.initTransactions();
 
       while (running) {
         ConsumerRecords<String, Order> records = consumer.poll(100);
         if (records.count() > 0) {
-          producer.beginTransaction();
+          // Commenting out configuration for Exactly Once Semantics (EOS)
+          // because currently EOS is not supported with Confluent Monitoring Interceptors 
+          //producer.beginTransaction();
           for (ConsumerRecord<String, Order> record : records) {
             Order order = record.value();
             if (OrderState.CREATED.equals(order.getState())) {
               //Validate the order then send the result (but note we are in a transaction so
               //nothing will be "seen" downstream until we commit the transaction below)
               producer.send(result(order, isValid(order) ? PASS : FAIL));
-              recordOffset(consumedOffsets, record);
+              // Commenting out configuration for Exactly Once Semantics (EOS)
+              // because currently EOS is not supported with Confluent Monitoring Interceptors 
+              //recordOffset(consumedOffsets, record);
             }
           }
-          producer.sendOffsetsToTransaction(consumedOffsets, CONSUMER_GROUP_ID);
-          producer.commitTransaction();
+          // Commenting out configuration for Exactly Once Semantics (EOS)
+          // because currently EOS is not supported with Confluent Monitoring Interceptors 
+          //producer.sendOffsetsToTransaction(consumedOffsets, CONSUMER_GROUP_ID);
```

Comment:
I think we should commit offset using the `KafkaConsumer` for non-EOS case?

Reply:
@mjsax , what do you think about:

```java
    if (eosEnabled) {
      consumerConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
    } else {
      consumerConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
    }
```

Reply:
Actually, modeling the suggestion from @bbejeck 

```java
    final String enableAutoCommit = eosEnabled ? "false" : "true";
    consumerConfig.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, enableAutoCommit);
```